run.sh

```diff
@@ -135,9 +144,22 @@ add_wildcard_record(){
   echo "address=/.${domain}/${hostmachineip}" > "/etc/dnsmasq.d/hostmachine.conf"
   echo -e "${GREEN}+ Added *.${domain} → ${hostmachineip}${RESET}"
 }
+setResolvConf(){
+  local tmpFile="${resolvConfigFile}.tmp"
+  
+  if [[ -f $resolvConfigFile ]]; then
+    cat $resolvConfigFile > $tmpFile
+    sed -i "/$resolvComment/d" $tmpFile
+    local localIp=$(hostname -i)
+    sed -i "1i nameserver $localIp $resolvComment" $tmpFile
+    cat $tmpFile > $resolvConfigFile
```

Comment:
What's the reason for going via this `$tmpfile`? Why not just modify `$resolvConfigFile` directly?

Reply:
Hmm.. I don't think I understand your worry about the lock?

Why not simply do

```bash
grep -qv "$resolvComment" "$resolvConfigFile" && sed -i "1s/^/nameserver ${localIp} ${resolvComment}\n/" "$resolvConfigFile"
```

This has some limitations though. Like if cleanup breaks, and the IP of devdns changes upon next start, this would fail to add the new IP.

Reply:
great, if it also solves that problem well I can use this approach :+1: